pkg/utils/release/controller.go

```diff
@@ -121,17 +115,160 @@ func (s *SuiteController) GetRelease(releaseName, releaseNamespace string) (*v1a
+				},
+			},
+		}
+	}
+	return releasePlan, s.KubeRest().Create(context.TODO(), releasePlan)
+}
+
+// GetReleasePlan returns the releasePlan with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlan(releasePlanName, devNamespace string) (*appstudiov1alpha1.ReleasePlan, error) {
+	releasePlan := &appstudiov1alpha1.ReleasePlan{}
+
+	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
+		Name:      releasePlanName,
+		Namespace: devNamespace,
+	}, releasePlan)
+
+	return releasePlan, err
+}
+
+//  specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted)
+func (s *SuiteController) DeleteReleasePlan(releasePlanName, devNamespace string, reportErrorOnNotFound bool) error {
+	releasePlan := &appstudiov1alpha1.ReleasePlan{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      releasePlanName,
+			Namespace: devNamespace,
+		},
+	}
+	err := s.KubeRest().Delete(context.TODO(), releasePlan)
+	if err != nil && !reportErrorOnNotFound && k8sErrors.IsNotFound(err) {
+		err = nil
+	}
+	return err
+}
+
+// CreateReleasePlanAdmission creates a new ReleasePlanAdmission using the given parameters.
+func (s *SuiteController) CreateReleasePlanAdmission(name, devNamespace, application, managedNamespace, environment, AutoReleaseLabel string, releaseStrategy string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
+	var releasePlanAdmission *appstudiov1alpha1.ReleasePlanAdmission
+
+	if AutoReleaseLabel != "" { // AutoReleaseLabel in ReleasePlanAdmission is not missing
+		releasePlanAdmission = &appstudiov1alpha1.ReleasePlanAdmission{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: managedNamespace,
+				Labels: map[string]string{
+					appstudiov1alpha1.AutoReleaseLabel: AutoReleaseLabel,
+				},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanAdmissionSpec{
+				DisplayName: name,
+				Application: application,
+				Origin: kcp.NamespaceReference{
+					Namespace: devNamespace,
+				},
+				Environment:     environment,
+				ReleaseStrategy: releaseStrategy,
+			},
+		}
+	} else { // AutoReleaseLabel in ReleasePlanAdmission is missing
+		releasePlanAdmission = &appstudiov1alpha1.ReleasePlanAdmission{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: managedNamespace,
+				Labels:    map[string]string{},
+			},
+			Spec: appstudiov1alpha1.ReleasePlanAdmissionSpec{
+				DisplayName: name,
+				Application: application,
+				Origin: kcp.NamespaceReference{
+					Namespace: devNamespace,
+				},
+				Environment:     environment,
+				ReleaseStrategy: releaseStrategy,
+			},
+		}
+	}
+	return releasePlanAdmission, s.KubeRest().Create(context.TODO(), releasePlanAdmission)
+}
+
+// CreateRegistryJsonSecret creates a secret for registry repositry in namespace given with key passed
+func (s *SuiteController) CreateRegistryJsonSecret(name string, namespace string, authKey string, keytName string) (*corev1.Secret, error) {
```

Comment:
```suggestion
func (s *SuiteController) CreateRegistryJsonSecret(name string, namespace string, authKey string, keyName string) (*corev1.Secret, error) {
```

Reply:
this shouldn't be resolved, and you can just put `string` once